src/types/mod.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod mongo;
+
+use std::{
+    fmt::Display,
+    ops::{
+        Deref,
+        DerefMut,
+    },
+    str::FromStr,
+};
+
+use anyhow::*;
+use bee_message_shimmer::semantic::ConflictReason;
+use derive_more::From;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, From, Hash, Ord, PartialOrd)]
+pub enum MessageId {
+    /// Chrysalis compatible message
+    Chrysalis(bee_message_cpt2::MessageId),
+    /// Shimmer compatible message
+    Shimmer(bee_message_shimmer::MessageId),
```

Comment:
Maybe we go with Stardust for now?

```suggestion
    Stardust(bee_message_shimmer::MessageId),
```

Reply:
🤷

Reply:
First, there will be Stardust, then Shimmer will come later and might contain more changes to the message layouts.